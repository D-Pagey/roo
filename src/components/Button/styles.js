import styled from 'styled-components';

export const Button = styled.button`
    background-color: #ffffff;
    border-radius: 3px;
    border: 0;
    color: #00a396;
    cursor: pointer;
    font-size: 1rem;
    padding: 11px 1rem;
`;
