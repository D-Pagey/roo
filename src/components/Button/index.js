import React from 'react';
import { string } from 'prop-types';
import * as S from './styles';

const Button = ({ children }) => <S.Button type="button">{children}</S.Button>;

Button.propTypes = {
    children: string.isRequired
};

export default Button;
