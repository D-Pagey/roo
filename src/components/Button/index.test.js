import React from 'react';
import { render } from '@testing-library/react';
import Button from '.';

describe('Button component', () => {
    it('should render', () => {
        const { container } = render(<Button>Change location</Button>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
