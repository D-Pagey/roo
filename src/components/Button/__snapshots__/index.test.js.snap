// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 3px;
  border: 0;
  color: #00a396;
  cursor: pointer;
  font-size: 1rem;
  padding: 11px 1rem;
}

<button
  class="c0"
  type="button"
>
  Change location
</button>
`;
